### PR TITLE
Optimizations

### PR DESCRIPTION
**Pull request by Carterpersall**
The intent of this pull request is to speed up winfetch wherever I can

Done:
- Optimized GPU fetching
  - Instead of using Get-CimInstance to get GPU information, which can be very slow, gets the registry entries at 'HKLM:\SOFTWARE\Microsoft\DirectX\', which contains a list of video adapters
  - 1690 ms -> 40.16 ms
    - 42x faster (nice)
- Made info_locale speedy
  - Takes the registy entries at HKCU:Control Panel\International to get region code and language code, and passes them into two very large hash tables to get the accompanying region and language values.
  - 3251 ms -> 12.43 ms
    - 262x Faster

TODO:
- [ ] Attempt to optimize expensive functions
  - [ ] info_ps_pkgs
    - 6630 ms
  - [ ] info_pkgs
    - 2400 ms
  - [x] info_gpu
    - 1640 ms -> 40.16 ms
      - 42x faster (nice)
    - [ ] Maybe: Fix edge case for users with multiple of the same GPU
  - [ ] info_cpu_usage
    - 1100 ms
  - [x] info_locale
    - 3251 ms -> 12.43 ms
      - 262x Faster
    - [ ] Make work when user has more than one language
    - [ ] Maybe: Find a way to circumvent the very large table
  - [ ] info_weather
    - 433 ms
  - [ ] info_local_ip
    - 1960 ms

### CODE REVIEW

**Comment by rashil2000** [comment]
@HO-COOH does this work as before with your setup (multiple GPUs)?

**Comment by jcwillox** [comment]
This PR currently returns GPUs which are no longer connected to the system, perhaps you could use the `LastSeen` registry key to identify old GPUs, but that might not be super reliable.

image

**Comment by Carterpersall** [comment]
> This PR currently returns GPUs which are no longer connected to the system, perhaps you could use the `LastSeen` registry key to identify old GPUs, but that might not be super reliable.
> 
> image

I was worried that could happen, I'll see what I can figure out.

**Comment by Carterpersall** [comment]
@jcwillox might be fixed in the most recent commit, can you check?

The LastSeen implementation should be reliable as the base DirectX key has a LastSeen value which can be compared with it's subkeys to see which were detected in its most recent run-through. This might also allow me to fix an edge case where if someone has two of the same GPU, it would be detected as a duplicate entry and not be shown.

**Comment by Carterpersall** [comment]
> @HO-COOH does this work as before with your setup (multiple GPUs)?

Works on my laptop with its integrated and dedicated graphics:

image

**Comment by jcwillox** [comment]
> @jcwillox might be fixed in the most recent commit, can you check?
> 
> The LastSeen implementation should be reliable as the base DirectX key has a LastSeen value which can be compared with it's subkeys to see which were detected in its most recent run-through. This might also allow me to fix an edge case where if someone has two of the same GPU, it would be detected as a duplicate entry and not be shown.

I tested it and it, unfortunately, returns no GPUs as the `HKLM:\SOFTWARE\Microsoft\DirectX\LastSeen` key doesn't seem to exist on my system.

image

File: winfetch.ps1 [hunk 1]
```diff
@@ -963,22 +966,365 @@ function info_battery {
...
+        "10039882" = "Northern Europe"
+        "10039883" = "Southern Africa"
+        "10210824" = "Western Europe"
+        "10210825" = "Australia and New Zealand"
+        "161832015" = "Saint Barthélemy"
+        "161832256" = "U.S. Minor Outlying Islands"
+        "161832257" = "Latin America and the Caribbean"
+        "161832258" = "Bonaire, Sint Eustatius and Saba"
+    }
+    $languageLookup = @{
```
Comment (rashil2000) [inline winfetch.ps1:1272]: This does not cover all languages (`English (India)` in my case)

Comment (Carterpersall) [reply winfetch.ps1:1272]: It seems that the languageLookup array wasn't complete. I'll fix that in my next commit.

**Comment by rashil2000** [comment]
Also, it would be better to split out optimizations for different functions in different PRs.

**Comment by Carterpersall** [comment]
Pull request moved to #153 ,#154, and #155

**Pull request closed without merging**